node/src/components/deploy_acceptor.rs

```diff
@@ -57,36 +55,21 @@ impl<REv> ReactorEventT for REv where
 {
 }
 
-#[derive(Debug, Clone, Serialize)]
-pub struct DeployAcceptorChainspec {
-    chain_name: String,
-    deploy_config: DeployConfig,
-}
-
-impl From<Chainspec> for DeployAcceptorChainspec {
-    fn from(chainspec: Chainspec) -> Self {
-        DeployAcceptorChainspec {
-            chain_name: chainspec.network_config.name,
-            deploy_config: chainspec.deploy_config,
-        }
-    }
-}
-
 /// The `DeployAcceptor` is the component which handles all new `Deploy`s immediately after they're
 /// received by this node, regardless of whether they were provided by a peer or a client.
 ///
 /// It validates a new `Deploy` as far as possible, stores it if valid, then announces the newly-
 /// accepted `Deploy`.
 #[derive(Debug)]
 pub struct DeployAcceptor {
-    cached_deploy_configs: HashMap<Version, DeployAcceptorChainspec>,
+    chainspec: Arc<Chainspec>,
```

Comment:
I see what you're trying to do here but the point of having separate structs was not only to save memory but to pass on only what the component needs to know, not everything.

Reply:
Fair point.  Would you be ok if I just add `chain_name` and `deploy_config` as members of `DeployAcceptor` rather than creating a whole new struct for them?